File: org.palladiosimulator.blockchainsystems.plugin/src/org/palladiosimulator/blockchainsystems/plugin/results/ui/ResultsRepositoryContainer.java
package org.palladiosimulator.blockchainsystems.plugin.results.ui;

import java.util.List;
import java.io.File;
import java.util.ArrayList;

public class ResultsRepositoryContainer {
	private List<ResultsRepository> _repositories;
	
	public ResultsRepositoryContainer() {
		_repositories = new ArrayList<ResultsRepository>();
	}
	
	public List<ResultsRepository> getResultRepositories() {
		return _repositories;
	}
	
	public void addRepository(File repositoryPath) {
		_repositories.add(new ResultsRepository(repositoryPath, this));
	}
}

File: org.palladiosimulator.blockchainsystems.plugin/src/main/kotlin/org/palladiosimulator/blockchainsystems/plugin/results/ui/ResultsRepositoryContainer.kt
package org.palladiosimulator.blockchainsystems.plugin.results.ui

import java.io.File

/**
 * Container for managing multiple results repositories.
 * Provides methods to add and retrieve repositories.
 *
 * @author Yannik Sproll, Davis Riedel
 */
class ResultsRepositoryContainer {
  val resultRepositories: MutableList<ResultsRepository> = ArrayList()

  fun addRepository(repositoryPath: File) {
    resultRepositories.add(ResultsRepository(repositoryPath, this))
  }
}